### PR TITLE
CEDS-1875 Use explicit Europe/London time zone

### PR DESCRIPTION
**Pull request by bambuchaAdm**

### CODE REVIEW

File: app/Module.scala [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import java.time.ZoneId
+
+import com.google.inject.AbstractModule
+
+class Module extends AbstractModule {
```
Comment (MaciejRewera) [inline app/Module.scala]: There is `modules` package. Please use it.

File: app/forms/ArrivalDetails.scala [hunk 2]
```diff
@@ -25,6 +27,6 @@ case class ArrivalDetails(dateOfArrival: Date, timeOfArrival: Time)
 object ArrivalDetails {
   implicit val format: OFormat[ArrivalDetails] = Json.format[ArrivalDetails]
 
-  val mapping: Mapping[ArrivalDetails] =
+  def mapping(zoneId: ZoneId): Mapping[ArrivalDetails] =
```
Comment (PatrykRudnicki) [inline app/forms/ArrivalDetails.scala]: Why this def has `zoneId? Why this `zoneId` is not used? And why it's `def`?

File: app/modules/DateTimeModule.scala [hunk 3]
```diff
@@ -20,15 +20,19 @@ import java.time.ZoneId
...
 import com.google.inject.AbstractModule
-import modules.DateTimeFormatterModule.NotificationsPageFormatter
+import javax.inject.{Inject, Provider, Singleton}
 
-class DateTimeFormatterModule extends AbstractModule {
-  override def configure(): Unit =
-    bind(classOf[DateTimeFormatter]).toInstance(NotificationsPageFormatter)
-}
+class DateTimeModule extends AbstractModule {
+  def timezone = ZoneId.of("Europe/London")
```
Comment (PatrykRudnicki) [inline app/modules/DateTimeModule.scala]: Why it's `def` not `val`?

File: app/services/audit/AuditService.scala [hunk 4]
```diff
@@ -32,7 +32,10 @@ import uk.gov.hmrc.play.audit.model.{DataEvent, ExtendedDataEvent}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class AuditService @Inject()(connector: AuditConnector, @Named("appName") appName: String)(implicit ec: ExecutionContext) {
+class AuditService @Inject()(
+                              connector: AuditConnector,
+                              @Named("appName") appName: String)
+                            (implicit ec: ExecutionContext) {
```
Comment (PatrykRudnicki) [inline app/services/audit/AuditService.scala]: Please use scalafmt

File: test/models/viewmodels/notificationspage/converters/ControlResponseBlockedConverterSpec.scala [hunk 5]
```diff
@@ -38,7 +39,11 @@ class ControlResponseBlockedConverterSpec extends BaseSpec with MockitoSugar {
     val decoder: Decoder = mock[Decoder]
     when(decoder.error(anyString)).thenReturn(Some(ILEError("CODE", "Messages.Key")))
 
-    val converter = new ControlResponseBlockedConverter(decoder, NotificationsPageFormatter)
+    private val injector = Guice.createInjector(new DateTimeTestModule(), new AbstractModule {
+      override def configure(): Unit = bind(classOf[Decoder]).toInstance(decoder)
+    })
+
+    val converter = injector.getInstance(classOf[ControlResponseBlockedConverter])
```
Comment (PatrykRudnicki) [inline test/models/viewmodels/notificationspage/converters/ControlResponseBlockedConverterSpec.scala]: Why those changes?

File: test/models/viewmodels/notificationspage/converters/ControlResponseRejectedConverterSpec.scala [hunk 6]
```diff
@@ -38,7 +39,11 @@ class ControlResponseRejectedConverterSpec extends BaseSpec with MockitoSugar {
     val decoder: Decoder = mock[Decoder]
     when(decoder.error(anyString)).thenReturn(Some(ILEError("CODE", "Messages.Key")))
 
-    val converter = new ControlResponseRejectedConverter(decoder, NotificationsPageFormatter)
+    private val injector = Guice.createInjector(new DateTimeTestModule, new AbstractModule {
+      override def configure(): Unit = bind(classOf[Decoder]).toInstance(decoder)
+    })
+
+    val converter = injector.getInstance(classOf[ControlResponseRejectedConverter])
```
Comment (PatrykRudnicki) [inline test/models/viewmodels/notificationspage/converters/ControlResponseRejectedConverterSpec.scala]: The same here

File: test/utils/DateTimeTestModule.scala [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+ */
+
+package utils
+
+import java.time.ZoneId
+
+import modules.DateTimeModule
+
+class DateTimeTestModule extends DateTimeModule {
+  override def timezone: ZoneId = ZoneId.of("UTC")
```
Comment (PatrykRudnicki) [inline test/utils/DateTimeTestModule.scala]: Why tests are using different zone?

File: app/modules/DateTimeModule.scala (hunk 3 above)
Comment (bambuchaAdm) [reply app/modules/DateTimeModule.scala]: for override - this is called only one on injector configuration.

File: test/utils/DateTimeTestModule.scala (hunk 7 above)
Comment (bambuchaAdm) [reply test/utils/DateTimeTestModule.scala]: They were written in different zone

**Pull request merged**